src/CoreWCF.MSMQ/src/CoreWCF/Channels/MsmqPoisonMessageException.cs

```diff
@@ -0,0 +1,18 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System;
+
+namespace CoreWCF.Channels
+{
+    public class MsmqPoisonMessageException : Exception
+    {
```

Comment:
This should be derivied from PoisonMessageException which derives from CommunicationException. All CoreWCF types ultimately derive from CommunicationException as it makes exception handling easier.